connector/src/main/java/org/jboss/as/connector/subsystems/datasources/AbstractDataSourceAdd.java

```diff
@@ -76,82 +77,91 @@
-
-        AbstractDataSourceService dataSourceService = createDataSourceService(dsName, jndiName);
-
-        final ManagementResourceRegistration registration = context.getResourceRegistrationForUpdate();
-        final ServiceName dataSourceServiceNameAlias = AbstractDataSourceService.SERVICE_NAME_BASE.append(jndiName);
-        final ServiceName dataSourceServiceName = context.getCapabilityServiceName(Capabilities.DATA_SOURCE_CAPABILITY_NAME, dsName, DataSource.class);
-        final ServiceBuilder<?> dataSourceServiceBuilder =
-                Services.addServerExecutorDependency(
-                        serviceTarget.addService(dataSourceServiceName, dataSourceService),
-                        dataSourceService.getExecutorServiceInjector(), false)
-                        .addAliases(dataSourceServiceNameAlias)
-                .addDependency(ConnectorServices.MANAGEMENT_REPOSITORY_SERVICE, ManagementRepository.class,
-                        dataSourceService.getManagementRepositoryInjector())
-                .addDependency(SubjectFactoryService.SERVICE_NAME, SubjectFactory.class,
-                        dataSourceService.getSubjectFactoryInjector())
-                .addDependency(ConnectorServices.JDBC_DRIVER_REGISTRY_SERVICE, DriverRegistry.class,
-                        dataSourceService.getDriverRegistryInjector())
-                .addDependency(ConnectorServices.IDLE_REMOVER_SERVICE)
-                .addDependency(ConnectorServices.CONNECTION_VALIDATOR_SERVICE)
-                .addDependency(ConnectorServices.IRONJACAMAR_MDR, MetadataRepository.class, dataSourceService.getMdrInjector())
-                .addDependency(ConnectorServices.RA_REPOSITORY_SERVICE, ResourceAdapterRepository.class, dataSourceService.getRaRepositoryInjector())
-                .addDependency(ConnectorServices.BOOTSTRAP_CONTEXT_SERVICE.append(DEFAULT_NAME))
-                .addDependency(NamingService.SERVICE_NAME);
-        if (jta) {
-            dataSourceServiceBuilder.addDependency(ConnectorServices.TRANSACTION_INTEGRATION_SERVICE, TransactionIntegration.class, dataSourceService.getTransactionIntegrationInjector())
-                    .addDependency(ConnectorServices.CCM_SERVICE, CachedConnectionManager.class, dataSourceService.getCcmInjector());
 
-        }
-        //Register an empty override model regardless of we're enabled or not - the statistics listener will add the relevant childresources
-        if (registration.isAllowsOverride()) {
-            registration.registerOverrideModel(dsName, DataSourcesSubsystemProviders.OVERRIDE_DS_DESC);
-        }
-        startConfigAndAddDependency(dataSourceServiceBuilder, dataSourceService, dsName, serviceTarget, operation);
-
-        dataSourceServiceBuilder.addDependency(driverServiceName, Driver.class,
+        final boolean enabled = ! operation.hasDefined(ENABLED.getName()) || ENABLED.resolveModelAttribute(context, model).asBoolean();
```

Comment:
The default value of ENABLED is 'true' so the "! operation.hasDefined(ENABLED.getName()) || " bit can be removed.